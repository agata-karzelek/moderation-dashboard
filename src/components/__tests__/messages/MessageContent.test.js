import React from "react";
import { shallow } from "enzyme";
import MessageContent from "../../messages/MessageContent";

const mockMsg = (text) => ({
  actions: text,
  text: "Delete it",
  file: {
    url: "https://abc.com",
  },
  reason: "Any Reason",
  moderatedMessage: "****",
  originalMessage: "shit",
  image: "",
  imageReason: ["Some Reason", "Other Reason"],
});
const mockDeleteFn = jest.fn();
const mockUpdateFn = jest.fn();
const mockUndoFn = jest.fn();

describe("Test Case For Message Content", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <MessageContent
        message={mockMsg("deleted")}
        toggledVal="chat"
        confirmDelete={mockDeleteFn}
        updateMessage={mockUpdateFn}
        confirmUndo={mockUndoFn}
        index={1}
      />
    );
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('check Functions with actions as "deleted"', () => {
    expect(mockUndoFn).toHaveBeenCalledTimes(0);
    wrapper.find("#undoIcon").simulate("click");
    expect(mockUndoFn).toHaveBeenCalledTimes(1);
  });

  test("check Functions with actions as empty", () => {
    wrapper = shallow(
      <MessageContent
        message={mockMsg("")}
        toggledVal="chat"
        confirmDelete={mockDeleteFn}
        updateMessage={mockUpdateFn}
        confirmUndo={mockUndoFn}
        index={1}
      />
    );

    expect(mockUpdateFn).toHaveBeenCalledTimes(0);
    wrapper.find("#updateIcon").simulate("click");
    expect(mockUpdateFn).toHaveBeenCalledTimes(1);

    expect(mockDeleteFn).toHaveBeenCalledTimes(0);
    wrapper.find("#deleteIcon").simulate("click");
    expect(mockDeleteFn).toHaveBeenCalledTimes(1);
  });
});
