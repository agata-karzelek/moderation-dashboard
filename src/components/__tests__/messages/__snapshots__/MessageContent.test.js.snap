// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case For Message Content Snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-messageContainer-22"
    container={true}
  >
    <span
      className="makeStyles-chat-4"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
        }
      }
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-deletedChat-21"
      >
        Delete it
      </WithStyles(ForwardRef(Typography))>
    </span>
    <Styled(MuiBox)
      ml={2}
    >
      <WithStyles(WithStyles(ForwardRef(Tooltip)))
        title="Undo"
      >
        <WithStyles(ForwardRef(IconButton))
          edge="start"
          id="undoIcon"
          onClick={[Function]}
        >
          <img
            alt="delete"
            src="/images/undo.svg"
          />
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(WithStyles(ForwardRef(Tooltip)))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
